numpy/ma/core.py

```diff
@@ -3859,7 +3859,8 @@ def _insert_masked_print(self):
 
                 rdtype = _replace_dtype_fields(self.dtype, "O")
                 res = data.astype(rdtype)
-                _recursive_printoption(res, mask, masked_print_option)
+                # XXX probably correct fix here is to modify copyto to accept scalars
+                _recursive_printoption(res, mask, np.array(masked_print_option, dtype='O'))
```

Comment:
`_recursive_printoption` calls `np.copyto(res, masked_print_option)`, which failed when numpy tried to convert `masked_print_option` to a 0d array in `PyArg_ParseTuple`.

Reply:
Oh, this fix probably isn't needed in this PR as-is. It is probably left over from when I also disallowed 0d object array creation.